File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/javascript/trie.js
/**
 * JavaScript Implementation: Trie (Prefix Tree)
 * Invariant: Prefix search O(Length of Word)
 */
export class TriePrefixTree {
  constructor() {
    this.items = [];
  }

  // TODO: Implement verified invariant
  insert(val) {
    this.items.push(val);
  }

  get size() {
    return this.items.length;
  }
}

File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/csharp/trie.cs
/**
 * C# Implementation: Trie (Prefix Tree)
 * Invariant: Prefix search O(Length of Word)
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class TriePrefixTree<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}
